sdk/rust/oak_derive/src/lib.rs

```diff
@@ -64,14 +64,20 @@ pub fn derive_oak_exports(input: TokenStream) -> TokenStream {
             // boundary, so catch any panics here and translate to an error return.
             // https://doc.rust-lang.org/nomicon/ffi.html#ffi-and-panics
             std::panic::catch_unwind(||{
-                let mut node = <#name>::new();
-                oak::grpc::event_loop(node, oak::ReadHandle{ handle: oak::Handle::from_raw(handle) })
+                oak::set_panic_hook();
+                main(handle)
             })
             .unwrap_or(Err(oak::OakStatus::ERR_INTERNAL))
             .err()
             .unwrap_or(oak::OakStatus::OK)
             .value()
         }
+        // Internal version of the main entrypoint, to allow testing without any
+        // panic interception.
+        pub fn main(handle: u64) -> Result<(), oak::OakStatus> {
```

Comment:
I think the term `main` is getting a bit overloaded here. There is `oak_main` and `main`, and we should probably refer to them as `entrypoint` anyways.

How about changing the derive macro to generate a `pub extern "C" fn entrypoint_#name`, and a `pub fn entrypoint_inner_#name` or similar? Or some variation of this.

Reply:
That's rather a bigger change than fits in this PR, because:

 - Having a varying name for the main entrypoint will remove the ability to have default configs (`oak/common/app_config.h`), which would in turn require changes to most of the examples.
 - The single-App server would immediately need a command-line option to specify the name of the initial Node entrypoint. 
 - (More minor) `#name` is a struct name so will lead to generated function names that clash with normal Rust style (e.g. `entrypoint_inner_Node`).

I've renamed `main` to `inner_main` for now; dropping the last two commits would also work.  

(If we choose to rename from "main" to "entrypoint", that can happen separately because it will again affect a lot more than this PR)